### DISCUSSION TITLE
CAD model import

### DISCUSSION THREAD

**Discussion by Bayarbadrakh** [idaholab/moose#17884] [Q&A Meshing] [answered]
Did anyone tried to import the mesh generated from the CAD software (Solidworks etc.) to MOOSE? Could share your experience on that?

**Comment by permcody** [comment, marked as answer]
I know we've had a few on the team import from Solidworks in the past. Maybe one or two will chime in. I've imported into Cubit myself and worked from CAD. What's nice about starting with CAD is that as you mesh you don't lose the underlying geometry (so if you change the resolution of your mesh, you'll continue to better capture curvature from your original CAD geometry. 

The one issue that many run into is that you may need to de-feature your CAD so that you can more easily mesh it. Very detailed geometry can be problematic for meshers. Cubit is one of the best packages available if you have access to it. I personally have no experience with other packages.

**Reply by makeclean** [reply]
+1 for Cubit - https://www.coreform.com/ - if you're doing academic things its free too. Nothing else I've tried really compares in terms of fixing and meshing. If your CAD is complex, you may want to consider ANSYS SpaceClaim which is literally unparalleled in its defeaturing and CAD repair. If you can get your geometry into a BREP form, then GMSH can pretty reliably produce meshes. However, if you've got complex CAD, then Cubit would be my go to too.
